src/arrayMethodSort.js

```diff
@@ -4,8 +4,28 @@
  * Implement method Sort
  */
 function applyCustomSort() {
-  [].__proto__.sort2 = function(compareFunction) {
-    // write code here
+  [].__proto__.sort2 = function (compareFunction) {
+    for (let i = 0; i < this.length; i++) {
+      for (let j = i; j < this.length; j++) {
+        if (compareFunction) {
+          if (compareFunction(this[i], this[j]) > 0) {
+            const temp = this[i];
+
+            this[i] = this[j];
+            this[j] = temp;
+          }
+        } else {
+          if (String(this[j]) < String(this[i])) {
```

Comment:
Be careful when converting elements to strings and comparing them. This approach will work for numbers and strings, but may not sort other data types like objects or undefined values as expected. Consider how you might handle different data types in a robust way.